### ISSUE TITLE
Params column position incorrect if placed on new line

### ISSUE THREAD

**Issue by ixje** [mgechev/revive#1020]
**Describe the bug**
**Background** 
The idea is to enforce a function signature style if the signature length exceeds a configured line width. i.e. assume this exceeds the limit
```golang
func boo(name1 string, name2 string, name3 bool, name4 []byte) (*[]byte, error) {
	return nil, nil
}
```
we want
```golang
func boo(
	name1 string,
	name2 string,
	name3 bool,
	name4 []byte,
) (*[]byte, error) {
	return nil, nil
}
```
**The bug**
Requesting the `Column` values for the parameters in the first code snippet works as expected.

Requesting the `Column` values when the parameters are on their own lines (as in the second snippet) are incorrect. Specifically they are off by 3 columns. This holds for all params. Requesting the `Column` values for something like `Results` is correct for both snippets. So this seems to be parameter specific.

**To Reproduce**
```shell
git clone -b column-issue https://github.com/ixje/revive.git
make build
./revive -config defaults.toml ./testdata/function-signature-style.go 
```
observe the output

**Expected behavior**
I expect all params to have a start column of `5` not of `2`

**Desktop (please complete the following information):**
 - OS: [e.g. Ubuntu 18.04]: fedora 40
 - Version of Go: go version go1.22.5 linux/amd64

**Comment by chavacava** [comment]
Hi @ixje, thanks for your interest in revive.

I've followed your instructions and I've got the following output

```
arg `name1` line: 8 col start: 5 col end: 17
arg `name2` line: 9 col start: 2 col end: 14
arg `name3` line: 10 col start: 2 col end: 12
arg `name4` line: 11 col start: 2 col end: 14
Results line: 12 col start: 3 col end: 19
```

IMO the column information is correct. The [definition](https://cs.opensource.google/go/go/+/go1.22.6:src/go/token/position.go;l=21) of field `Position.Column` specifies that the value represents the _character count_ 
```go
type Position struct {
    Filename string // filename, if any
    Offset   int    // offset, starting at 0
    Line     int    // line number, starting at 1
    Column   int    // column number, starting at 1 (character count)
}
```

In the test file, the field names start at the second (2) character of the corresponding lines (the first character being a `\t` (tabulation) character)

I hope this helps, if is it the case please feel free to close the ticket.
BTW, if you plan to propose a new rule, please open an issue to discuss about it *before* making any PR.

**Comment by ixje** [comment]
That's odd I get this as result (note the difference in arg `name1` column value)
```
arg `name1` line: 8 col start: 2 col end: 14
arg `name2` line: 9 col start: 2 col end: 14
arg `name3` line: 10 col start: 2 col end: 12
arg `name4` line: 11 col start: 2 col end: 14
Results line: 12 col start: 3 col end: 19
```


This is what my editor (Goland) shows
![column](https://github.com/user-attachments/assets/c5c7000c-27be-44cf-afe1-c5caa2a9ff16)
Is there a way to get this column count (`5`) instead? 
I understand there is a `\t` before the params, but I'm not sure how to detect that as I'd like to provide the correct column in the error response as to what the user will look for.

I don't think this is an editor issue either as both VSCode and Sublime also think this is column 5
![Screenshot from 2024-08-09 22-06-56](https://github.com/user-attachments/assets/79bb3538-74c1-4bc6-9040-4d6666b55685)

Would love to hear any tips/hints how deal with this issue. Thanks.

**Comment by chavacava** [comment]
> That's odd I get this as result (note the difference in arg name1 column value)

My bad, the output I've mentioned is for a modified test file. The initial output was exactly the one you provided.

About editors: editors "interpret" tabulation as a sequence of spaces. That's why vscode is able to show "col 5" in your screenshot.
Try to make the following experiment: On vscode modify the settings for "Tab size" by setting it to something like 7. You will see field names starting at column 8.

**Comment by ixje** [comment]
I'll close this as this seems to be by design. I found I can fix the column by detecting the `\t` through the `file.Content()` and `Position.Offset` and adjusting the `Column` when error reporting. 

Thanks for the insights
